pyccl/halos/profiles.py

```diff
@@ -147,6 +147,38 @@ def _get_plaw_projected(self, cosmo, a):
         """
         return self.precision_fftlog['plaw_projected']
 
+    def __eq__(self, prof2):
+        """ Return `True` if this profile is equivalent to another."""
+        if type(self) != type(prof2):
+            # catch different profile types
+            return False
+
+        params, params2 = self.__dict__, prof2.__dict__
+
+        if params == params2:
+            # same profiles
+            return True
+
+        for key, val in params.items():
+            if key == "cM":
+                cm2 = params2.get(key)
+                if val.name != cm2.name:
+                    return False
+                if not val.mdef.__eq__(cm2.mdef):
+                    return False
```

Comment:
OK, this is all good as long as c-M relations are fully defined by their name and mass definition. However, for instance the Ishiyama model you implemented also depends on whether you pick relaxed halos or use the Vmax method. E.g. you could have two profiles generated with the same Ishiyama cM relation and mass definition, but one for relaxed and the other for all halos, and this would identify both profiles as the same, right?

Reply:
I see your point. There are two alternatives we can choose from:

1. Either I include a `__dict__` check in the statement so that it also checks the other attributes on top of the mass definition;
2. Or I delete the long for-loop-check altogether and if the two `cM` relations are not identical, it returns False.

These are all edge cases, so I don't think it will make a difference no matter which one we choose. 1 is more complete, 2 is simpler.

Reply:
OK, I have another proposal, which I'll leave in the main comment for this review.